### PR TITLE
[ADLA] - Support catalog list table fragments

### PR DESCRIPTION
**Pull request by idear1203**
### Swagger PR Links
- [Azure/azure-rest-api-specs#2869](https://github.com/Azure/azure-rest-api-specs/pull/2869)
- [Azure/azure-rest-api-specs#2944](https://github.com/Azure/azure-rest-api-specs/pull/2944)

### Changes in 3.4.0-preview
- [Azure/azure-rest-api-specs#2869](https://github.com/Azure/azure-rest-api-specs/pull/2869)
    1. There are no functional changes here just code refactor to make it more readable
- [Azure/azure-rest-api-specs#2944](https://github.com/Azure/azure-rest-api-specs/pull/2944)
   1. Regenerate ADLA SDK to support list table fragments. Relevant files include:
       - [`CatalogOperations.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Generated/ICatalogOperations.cs)
      - [`CatalogOperationsExtensions.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Generated/CatalogOperationsExtensions.cs)
       - [`ICatalogOperations.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Generated/ICatalogOperations.cs)
      - New file:  [`USqlTableFragment.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Generated/Models/USqlTableFragment.cs)
  2. Add test for the included change. Relevant files include:
      - [`DataLakeAnalyticsManagementHelper.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/TestHelpers/DataLakeAnalyticsManagementHelper.cs): Update script to create table fragments
      - [`CatalogOperationTests.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70/src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/ScenarioTests/CatalogOperationTests.cs): Add test for listing table fragments
     - Remove the following folder since they are obsolete and already has replacement now.
       - [`DataLakeAnalyticsCatalog.Tests.CatalogOperationTests`](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/SessionRecords/DataLakeAnalyticsCatalog.Tests.CatalogOperationTests)
       - [`DataLakeAnalyticsCatalog.Tests.CommonTestFixture`](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/SessionRecords/DataLakeAnalyticsCatalog.Tests.CommonTestFixture)
       - [`DataLakeAnalyticsJob.Tests.JobOperationTests`](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/SessionRecords/DataLakeAnalyticsJob.Tests.JobOperationTests)
     - Update [`SessionRecords`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/DataLake.Analytics/DataLakeAnalytics.Tests/SessionRecords)
    3. Update version. Relevant files include:
       - [`AssemblyInfo.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Properties/AssemblyInfo.cs)
       - [`changelog.md`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/DataLake.Analytics/changelog.md)
       - [`DataLakeAnalyticsCustomizationHelper.cs`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs)
        - [`datalake-analytics_data-plane.txt`](//src/SDKs/_metadata/datalake-analytics_data-plane.txt)
        - [`datalake-analytics_resource-manager.txt`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/_metadata/datalake-analytics_resource-manager.txt)
        - [`Microsoft.Azure.Management.DataLake.Analytics.csproj`](https://github.com/Azure/azure-sdk-for-net/blob/ce3bde7ac2d10eb631031a4e1c7ef3f99be3ce70//src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics.csproj)

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by ro-joowan — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]
LGTM

File: src/SDKs/_metadata/datalake-analytics_data-plane.txt [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 1) azure-rest-api-specs repository information
 GitHub user: Azure
 Branch:      master
-Commit:      f6330512616c264543fec0f02f21072a1559173b
+Commit:      2ae4511d8575f4f9efae0b0c144ef4f5adc8546f
 
 2) AutoRest information
 Requested version: latest
-Bootstrapper version:    C:\Users\jooro\AppData\Roaming\npm `-- autorest@2.0.4245  
+Bootstrapper version:    D:\code\nodejs\node_global `-- autorest@2.0.4262
```
Comment (dsgouda) [inline src/SDKs/_metadata/datalake-analytics_data-plane.txt]: Not sure why this file is being updated, @idear1203 which `generate.cmd` was run to generate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please run `msbuild build.proj /t:build /p:Scope=SDKs\DataLake.Analytics`
and commit any changes to the .props file

**Comment by dsgouda** [comment]
Please update the branch with upstream

File: src/SDKs/_metadata/datalake-analytics_data-plane.txt (hunk 1 above)
Comment (idear1203) [reply src/SDKs/_metadata/datalake-analytics_data-plane.txt]: I use this `generate.cmd`: [src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/generate.cmd](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/DataLake.Analytics/Management.DataLake.Analytics/generate.cmd).

This file seems updating frequently.

**Comment by idear1203** [comment]
The branch has been updated with upstream as request.

**Comment by idear1203** [comment]
The branch has been updated with upstream as request.

---
In reply to: [387803603](https://github.com/Azure/azure-sdk-for-net/pull/4276#issuecomment-387803603) [](ancestors = 387803603)

**Comment by idear1203** [comment]
I have run the command `msbuild build.proj /t:build /p:Scope=SDKs\DataLake.Analytics`, but I cannot see any new changes.

BTW, it seems there is no such `.props` file under `src\SDKs\DataLake.Analytics`.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**